### PR TITLE
Increase code coverage (pt2).

### PR DESCRIPTION
**Pull request by Laimiux**

### CODE REVIEW

File: formula/src/main/java/com/instacart/formula/internal/ChildrenManager.kt [hunk 1]
```diff
@@ -76,38 +75,27 @@ internal class ChildrenManager(
                 )
             }
 
-            if (key is IndexedKey) {
```
Comment (Laimiux) [inline formula/src/main/java/com/instacart/formula/internal/ChildrenManager.kt:79]: Removing this as this will never happen

**Comment by carrotkite** [comment]
### JaCoCo Code Coverage 96.28% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|
| `com/instacart/formula/internal/Listeners` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/internal/ChildrenManager` | 100% | 0% | :white_check_mark: |
| `com/instacart/formula/internal/SingleRequestHolder` | 100% | 0% | :white_check_mark: |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: formula/src/main/java/com/instacart/formula/internal/ChildrenManager.kt [hunk 2]
```diff
@@ -12,7 +11,7 @@ internal class ChildrenManager(
     private val delegate: FormulaManagerImpl<*, *, *>,
     private val inspector: Inspector?,
 ) {
-    private var children: SingleRequestMap<Any, FormulaManager<*, *>>? = null
+    private val children: SingleRequestMap<Any, FormulaManager<*, *>> = LinkedHashMap()
```
Comment (Laimiux) [inline formula/src/main/java/com/instacart/formula/internal/ChildrenManager.kt:14]: No need to optimize `children` map initialization as `FormulaManager` optimizes this already

File: formula/src/main/java/com/instacart/formula/internal/Listeners.kt [hunk 3]
```diff
@@ -1,31 +1,20 @@
...
     private var indexes: MutableMap<Any, Int>? = null
 
-    private fun duplicateKeyErrorMessage(key: Any): String {
-        return "Listener $key is already defined. Unexpected issue."
-    }
-
     fun <Input, State, Event> initOrFindListener(key: Any, useIndex: Boolean): ListenerImpl<Input, State, Event> {
         val currentHolder = listenerHolder<Input, State, Event>(key)
-        return if (currentHolder.requested && useIndex) {
-            if (key is IndexedKey) {
```
Comment (Laimiux) [inline formula/src/main/java/com/instacart/formula/internal/Listeners.kt:16]: This won't happen so no need to test

**Review by deesonpatel — APPROVED** [review]

**Pull request merged**